GitCommands/Git/WorkTree/GitWorkTree.cs

```diff
@@ -0,0 +1,19 @@
+namespace GitCommands.Git.WorkTree
+{
+    public class GitWorkTree
+    {
+        public string? Path { get; set; }
+        public HeadType Type { get; set; }
+        public string? Sha1 { get; set; }
+        public string? CompleteBranchName { get; set; }
+        public string? BranchName { get; set; }
+        public bool IsDeleted { get; set; }
+    }
+
+    public enum HeadType
```

Comment:
1. OK
2. These types are mentioned in `git-worktree` docs. I don't now if it can be used in other parts or the repo. Now there are some methods in `GitModule` like `bool IsDetachedHead()` and `bool IsBareRepository()`

Reply:
> 2\. These types are mentioned in `git-worktree` docs.

Skimming the docs these didn't strike me as a set of explicit ref types, which are worktree specific.
But I now see this is an existing code moved around.